src/test/java/ResultsServletTest.java

```diff
@@ -0,0 +1,180 @@
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Arrays;
+import com.google.gson.Gson;
+
+@RunWith(JUnit4.class)
+public class ResultsServletTest {
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private DatastoreService testDatastore;
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+
+    testDatastore = DatastoreServiceFactory.getDatastoreService();
+
+    Entity recipeEntity1 = new Entity("Recipe");
+    List<String> searchStrings1 = new ArrayList<String>();
+    searchStrings1.add("EGG");
+    searchStrings1.add("FLOUR");
+    recipeEntity1.setProperty("search-strings", searchStrings1);
+
+    Entity recipeEntity2 = new Entity("Recipe");
+    List<String> searchStrings2 = new ArrayList<String>();
+    searchStrings2.add("EGG");
+    searchStrings2.add("BUTTER");
+    recipeEntity2.setProperty("search-strings", searchStrings2);
+
+    Entity recipeEntity3 = new Entity("Recipe");
+    List<String> searchStrings3 = new ArrayList<String>();
+    searchStrings3.add("AVOCADO");
+    searchStrings3.add("TOMATO");
+    recipeEntity3.setProperty("search-strings", searchStrings3);
+
+    testDatastore.put(recipeEntity1);
+    testDatastore.put(recipeEntity2);
+    testDatastore.put(recipeEntity3);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  /** Tests that doGet actually utilizes the user-query. */
+  @Test
+  public void doGetTest() throws Exception {
+    HttpServletRequest request = mock(HttpServletRequest.class);       
+    HttpServletResponse response = mock(HttpServletResponse.class);    
+
+    when(request.getParameter("user-query")).thenReturn("");
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+
+    new ResultsServlet().doGet(request, response);
+
+    verify(request, atLeast(1)).getParameter("user-query"); // Verify that user-query was called.
+    writer.flush();
+  }
+
+  /** Tests the basic functionality of doGet (i.e. logic of retrieving
+      results from Datastore based on user keyword input).
+      Not directly mocking the object because 1) Datastore is a final class
+      and 2) The following test ensures the same logic holds while using
+      the existing mocked Datastore service (testDatastore). */
+  @Test
+  public void doGetContentTest_singleKeyword() throws Exception {
+    List<String> eggQuery = new ArrayList<String>();
```

Comment:
Please consider above method of list initialization and in place without variable to avoid unnecessary verbosity.